compiler/transf.nim

```diff
@@ -324,6 +324,14 @@ proc introduceNewLocalVars(c: PTransf, n: PNode): PNode =
     if a.kind == nkSym:
       n[1] = transformSymAux(c, a)
     return n
+  of nkProcDef: # todo optimize nosideeffects?
+    result = newTransNode(n)
+    let x = freshVar(c, n[namePos].sym)
+    idNodeTablePut(c.transCon.mapping, n[namePos].sym, x)
+    result[namePos] = x # we have to copy proc definitions for iters
+    for i in 1..<n.len:
+      result[i] = introduceNewLocalVars(c, n[i])
+    result[namePos].sym.ast = result
```

Comment:
Did you try this instead?

```suggestion
  of nkProcDef, nkFuncDef, ...:
    result = n
```

Reply:
I tried templates which aren't affected by this bug. `nkFuncDef` should be fine since it cannot access outside variables. `method` is only allowed at top level.

Reply:
It doesn't seem to work.
```c
error: 'mX60gensym1___test_15' undeclared (first use in
this function); did you mean 'mX60gensym1___test_19'?
```